TaskModules/powershell/TaskModuleIISManageUtility/AppCmdOnTargetMachines.ps1

```diff
@@ -773,7 +801,7 @@ function Execute-Main
     {
         Add-And-Update-AppPool -appPoolName $AppPoolName -clrVersion $DotNetVersion -pipeLineMode $PipeLineMode -identity $AppPoolIdentity -appPoolCredentials $AppPoolCredentials 
     }
-    else {
+    elseif($ActionIISApplicationPool -eq $null) {

```

Comment:
why do this? Shouldn't the caller have sent appropriate value for this variable.